client/state/analytics/middleware.js

```diff
@@ -0,0 +1,40 @@
+import analytics from 'analytics';
+import invoke from 'lodash/invoke';
+import reduxAnalytics from 'redux-analytics';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants';
+
+const eventServices = {
+	ga: ( { category, action, label, value } ) => analytics.ga.recordEvent( category, action, label, value ),
+	tracks: ( { name, properties } ) => analytics.tracks.recordEvent( name, properties )
+};
+
+const pageViewServices = {
+	ga: ( { url, title } ) => analytics.ga.recordPageView( url, title ),
+	default: ( { url, title } ) => analytics.pageView.record( url, title )
+};
+
+const statBump = ( { group, name } ) => analytics.mc.bumpStat( group, name );
+
+export const dispatcher = ( { analytics: { type, payload } } ) => {
+	const { service = 'default' } = payload;
+
+	switch ( type ) {
+		case RECORD_EVENT:
+			return invoke( eventServices, service, payload );
+
+		case RECORD_PAGE_VIEW:
+			return invoke( pageViewServices, service, payload );
+
+		case BUMP_STAT:
+			return statBump( payload );
+	}
+};
+
+export const analyticsMiddleware = reduxAnalytics( dispatcher );
```

Comment:
I took a quick look at the library, and I'm also for not pulling in the dependency. The lib is tiny and writing the middleware would give us more freedom in terms of payload format.